geometry/eolearn/geometry/superpixel.py

```diff
@@ -0,0 +1,88 @@
+"""
+Module for super-pixel segmentation
+"""
+
+import logging
+
+import skimage.segmentation
+import numpy as np
+
+from eolearn.core import EOTask, FeatureType, FeatureTypeSet
+
+LOGGER = logging.getLogger(__name__)
+
+
+class SuperpixelSegmentation(EOTask):
+    """ Super-pixel segmentation task
+
+    Given a raster feature it will segment data into super-pixels. Representation of super-pixels will be returned as
+    a mask timeless feature where all pixels with the same value belong to one super-pixel
+    """
+    def __init__(self, feature, superpixel_feature, *, segmentation_object=skimage.segmentation.felzenszwalb,
+                 **segmentation_params):
+        """
+        :param feature: Raster feature which will be used in segmentation
+        :param superpixel_feature: A new mask timeless feature to hold super-pixel mask
+        :param segmentation_object: A function (object) which performs superpixel segmentation, by default that is
+            `skimage.segmentation.felzenszwalb`
+        :param segmentation_params: Additional parameters which will be passed to segmentation_object function
+        """
+        self.feature_checker = self._parse_features(feature, allowed_feature_types=FeatureTypeSet.SPATIAL_TYPES)
```

Comment:
Any reason why `feature_checker` is build in `__init__`, and then `next` is called to retrieve the feature only in `execute`, while `superpixel_feature` is constructed here?

Reply:
The idea is that you can use `FeatureParser` with or without having an EOPatch. If you parse with an EOPatch it will also make sure that parsed features exist in the EOPatch before returning them.

In this case the `feature` parameter is an input feature for which we have to check that it exists in any given EOPatches. So we collect parsed feature from `FeatureParser` only in `execute` method, when we have an EOPatch.
On the other hand `superpixel_feature` is a new feature which does not have to exist in EOPatches yet. So we can finish parsing the feature right away in the init.

I agree that the way this is done looks a bit weird. An for the future there is a plan to improve the way features are parsed, but that will have to be done in the core package.